package/kernel/lantiq/ltq-deu/src/ifxmips_aes.c

```diff
@@ -251,23 +299,24 @@ void ifx_deu_aes (void *ctx_arg, u8 *out_arg, const u8 *in_arg,
 
     /* To handle all non-aligned bytes (not aligned to 16B size) */
     if (byte_cnt) {
-        aes->ID3R = INPUT_ENDIAN_SWAP(*((u32 *) in_arg + (i * 4) + 0));
-        aes->ID2R = INPUT_ENDIAN_SWAP(*((u32 *) in_arg + (i * 4) + 1));
-        aes->ID1R = INPUT_ENDIAN_SWAP(*((u32 *) in_arg + (i * 4) + 2));
-        aes->ID0R = INPUT_ENDIAN_SWAP(*((u32 *) in_arg + (i * 4) + 3));    /* start crypto */
+        u8 temparea[16];
+
+        memcpy(temparea, ((u32 *) in_arg + (i * 4)), byte_cnt);
```

Comment:
`u8 temparea[16];` is used partially uninitialized here. 
If `byte_cnt` is not 16, it will not completely  fill the `temparea` array and some dta is uninitialized. 
This would leak some data from the kernel stack to the encrypted data.